src/atomate2/vasp/sets/base.py

```diff
@@ -968,59 +973,6 @@ def _remove_unused_incar_params(incar, skip=None):
                 incar.pop(ldau_flag, None)
 
 
-def _set_kspacing(
-    incar,
-    incar_settings,
-    user_incar_settings,
-    auto_kspacing,
-    bandgap,
-    kpoints,
-    from_prev,
-):
-    """
-    Set KSPACING in an INCAR.
-
-    if kpoints is not None then unset any KSPACING
-    if kspacing set in user_incar_settings then use that
-    if auto_kspacing then do that
-    if kspacing is set in config use that.
-    if from_prev is True, ISMEAR will be set according to the band gap
-    """
-    if kpoints is not None:
-        # unset KSPACING as we are using a KPOINTS file
-        incar.pop("KSPACING", None)
-
-        # Ensure adequate number of KPOINTS are present for the tetrahedron method
-        # (ISMEAR=-5). If KSPACING is in the INCAR file the number of kpoints is not
-        # known before calling VASP, but a warning is raised when the KSPACING value is
-        # > 0.5 (2 reciprocal Angstrom). An error handler in Custodian is available to
-        # correct overly large KSPACING values (small number of kpoints) if necessary.
-        if np.product(kpoints.kpts) < 4 and incar.get("ISMEAR", 0) == -5:
-            incar["ISMEAR"] = 0
-
-    elif "KSPACING" in user_incar_settings:
```

Comment:
@jmmshn Do you think these lines could be the cause of the test failures in #412 ?